pimd/pim6_cmd.c

```diff
@@ -388,6 +389,95 @@ DEFPY (show_ipv6_pim_neighbor_vrf_all,
 	return CMD_SUCCESS;
 }
 
+DEFPY (show_ipv6_pim_nexthop,
+       show_ipv6_pim_nexthop_cmd,
+       "show ipv6 pim [vrf NAME] nexthop",
+       SHOW_STR
+       IPV6_STR
+       PIM_STR
+       VRF_CMD_HELP_STR
+       "PIM cached nexthop rpf information\n")
+{
+	struct vrf *vrf_struct;
+
+	vrf_struct = vrf_lookup_by_name(vrf ? vrf : VRF_DEFAULT_NAME);
+
+	if (!vrf_struct)
+		return CMD_WARNING;
+
+	pim_show_nexthop(vrf_struct->info, vty);
+
+	return CMD_SUCCESS;
+}
+
+DEFPY (show_ipv6_pim_nexthop_lookup,
+       show_ipv6_pim_nexthop_lookup_cmd,
+       "show ipv6 pim [vrf NAME] nexthop-lookup [X:X::X:X$addr] [X:X::X:X$addr1]",
+       SHOW_STR
+       IPV6_STR
+       PIM_STR
+       VRF_CMD_HELP_STR
+       "PIM cached nexthop rpf lookup\n"
+       "Source/RP address\n"
+       "Multicast Group address\n")
+{
+	struct prefix nht_p;
+	int result = 0;
+	pim_addr src_addr, grp_addr;
+	pim_addr vif_source;
+	struct prefix grp;
+	struct pim_nexthop nexthop;
+	char nexthop_addr_str[PIM_ADDRSTRLEN];
+	char grp_str[PIM_ADDRSTRLEN];
+	struct vrf *vrf_struct;
+
+	vrf_struct = vrf_lookup_by_name(vrf ? vrf : VRF_DEFAULT_NAME);
+
+	if (!vrf_struct)
+		return CMD_WARNING;
+
+	result = inet_pton(AF_INET6, addr_str, &src_addr);
```

Comment:
This may not be required here since we are already getting the src_addr as addr itself, you can try running the command without this check and see if the CLI is validating this check already.